src/main/java/uk/gov/hmcts/reform/translate/errorhandling/RestExceptionHandler.java

```diff
@@ -0,0 +1,77 @@
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.method.annotation.MethodArgumentTypeMismatchException;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+import uk.gov.hmcts.reform.translate.model.TranslationsRequest;
+
+import java.io.Serializable;
+import javax.servlet.http.HttpServletRequest;
+import javax.validation.ConstraintViolationException;
+
+
+@Slf4j
+@ControllerAdvice
+public class RestExceptionHandler extends ResponseEntityExceptionHandler {
+
+    @ExceptionHandler({ApiException.class})
+    @ResponseBody
+    public ResponseEntity<HttpError<Serializable>> handleApiException(final HttpServletRequest request,
+                                                                      final Exception exception) {
+        log.error(exception.getMessage(), exception);
+        final HttpError<Serializable> error = new HttpError<>(exception, request);
+        return ResponseEntity
+            .status(error.getStatus())
+            .body(error);
+    }
+
+    @ExceptionHandler({
+        ConstraintViolationException.class,
+        MethodArgumentTypeMismatchException.class,
+        HttpMessageConversionException.class
+    })
+    @ResponseBody
+    public ResponseEntity<HttpError<Serializable>> handleCommonExceptionsAsBadRequest(final HttpServletRequest request,
+                                                                                      final Exception exception) {
+        log.error(exception.getMessage(), exception);
+        final HttpError<Serializable> error = new HttpError<>(exception, request, HttpStatus.BAD_REQUEST);
+        return ResponseEntity
+            .status(HttpStatus.BAD_REQUEST)
+            .body(error);
+    }
+
+    @Override
+    protected ResponseEntity<Object> handleMethodArgumentNotValid(MethodArgumentNotValidException exception,
+                                                                  HttpHeaders headers,
+                                                                  HttpStatus status,
+                                                                  WebRequest request) {
+        log.debug("MethodArgumentNotValidException:{}", exception.getLocalizedMessage());
+        final HttpError<Serializable> error = new HttpError<>(exception, request, HttpStatus.BAD_REQUEST)
+            .withMessage(TranslationsRequest.BAD_REQUEST_MESSAGE_BAD_SCHEMA);
+        return ResponseEntity
+            .status(HttpStatus.BAD_REQUEST)
+            .body(error);
+    }
+
+    @Override
+    protected ResponseEntity<Object> handleHttpMessageNotReadable(
+        HttpMessageNotReadableException exception, HttpHeaders headers, HttpStatus status, WebRequest request) {
+
+        log.debug("MethodArgumentNotValidException:{}", exception.getLocalizedMessage());
```

Comment:
Can we reword exception name in debug message to match trapped exception: `HttpMessageNotReadableException`

Reply:
Fixed in [6bbc198](https://github.com/hmcts/ts-translation-service/pull/51/commits/6bbc198413c4d31eff009bae0d5dc04f9ecf2407)